sklearn/feature_extraction/text.py

```diff
@@ -1099,6 +1107,10 @@ def idf_(self):
         # which means hasattr(self, "idf_") is False
         return np.ravel(self._idf_diag.sum(axis=0))
 
+    def _get_tags(self):
+        return _update_tags(self, super(TfidfTransformer, self),
+                            input_types=["sparse"])
```

Comment:
Originally this PR changed this, but it led to weirdness so I decided to leave it for now.